scipy/fft/_pocketfft/pocketfft_hdronly.h

```diff
@@ -149,7 +149,7 @@ template<> struct VLEN<double> { static constexpr size_t val=2; };
 #endif
 #endif
 
-#if __cplusplus >= 201703L
+#if __cplusplus >= 201703L && !defined(_MSVC_LANG)
```

Comment:
We're not using [`_MSVC_LANG`](https://github.com/search?q=repo%3Ascipy%2Fscipy%20%2F_MSVC_LANG%2F&type=code) elsewhere for this, so I'd like to keep this on [`_MSC_VER`](https://github.com/search?q=repo%3Ascipy%2Fscipy+%2F_MSC_VER%2F&type=code). Also, we should document the _why_, e.g.

```suggestion
// MSVC does not provide the standard aligned allocation functions,
// yet _still_ sets the feature test macro (__cpp_aligned_new)...
// While there are replacement functions (_aligned_malloc & _aligned_free),
// those produce crashes when used here, see GH-19726
#if __cplusplus >= 201703L && !defined(_MSC_VER)
```